framework/aster-frame/src/vm/frame.rs

```diff
@@ -723,17 +723,29 @@ impl<'a> VmWriter<'a> {
 
     /// Fills the available space by repeating `value`.
     ///
+    /// Returns the number of values written.
+    ///
     /// # Panic
     ///
     /// The size of the available space must be a multiple of the size of `value`.
     /// Otherwise, the method would panic.
-    pub fn fill<T: Pod>(&mut self, value: T) {
-        assert!(self.avail() / value.as_bytes().len() > 0);
-        assert!(self.avail() % value.as_bytes().len() == 0);
+    pub fn fill<T: Pod>(&mut self, value: T) -> usize {
+        assert!(self.avail() % core::mem::size_of::<T>() == 0);
```

Comment:
I think an assertion of `self.cursor % core::mem::size_of::<T>() == 0` should be added, which is one of the safety requirements of `ptr::write`.